www/input.js

```diff
@@ -0,0 +1,486 @@
+     */
+    _mouseButtonMovement(event) {
+        const down = (event.type === 'mousedown' ? 1 : 0);
+        var data = {};
+
+        if (event.type === 'mousemove' && !this.m) return;
+
+        if (!document.pointerLockElement) {
+            if (this.mouseRelative)
+                event.target.requestPointerLock();
+        }
+
+        // Hotkey to enable pointer lock, CTRL-SHIFT-LeftButton
+        if (down && event.button === 0 && event.ctrlKey && event.shiftKey) {
+            event.target.requestPointerLock();
+            return;
+        }
+
+        if (document.pointerLockElement) {
+            // FIXME - mark as relative!
+            console.warn("FIXME: Make event relative!")
+            this.x = event.movementX;
+            this.y = event.movementY;
+        } else if (event.type === 'mousemove') {
+            this.x = this._clientToServerX(event.clientX);
+            this.y = this._clientToServerY(event.clientY);
+            data["event"] = "MouseMove"
+        }
+
+        if (event.type === 'mousedown') {
+            data["event"] = "MouseButtonPress";
+        } else if (event.type === 'mouseup') {
+            data["event"] = "MouseButtonRelease";
+        }
+
+        if (event.type === 'mousedown' || event.type === 'mouseup') {
+            var mask = 1 << event.button;
+            if (down) {
+                this.buttonMask |= mask;
+            } else {
+                this.buttonMask &= ~mask;
+            }
+
+            data["button"] = this.buttonMask;
+        }
+
+        data["x"] = this.x;
+        data["y"] = this.y;
+
+        this.send(JSON.stringify(data));
+    }
+
+    /**
+     * Handles touch events and sends them to WebRTC app.
+     * @param {TouchEvent} event
+     */
+    _touch(event) {
+        var mask = 1;
+        var data = {};
+
+        if (event.type === 'touchstart') {
+            this.buttonMask |= mask;
+            data["event"] = "MouseButtonPress";
+            data["button"] = this.buttonMask;
+        } else if (event.type === 'touchend') {
+            this.buttonMask &= ~mask;
+            data["event"] = "MouseButtonRelease";
+            data["button"] = this.buttonMask;
+        } else if (event.type === 'touchmove') {
+            event.preventDefault();
+            data["event"] = "MouseMove";
+        }
+
+        this.x = this._clientToServerX(event.changedTouches[0].clientX);
+        this.y = this._clientToServerY(event.changedTouches[0].clientY);
+
+        data["x"] = this.x;
+        data["y"] = this.y;
+
+        this.send(JSON.stringify(data));
+    }
+
+    /**
+     * Handles mouse wheel events and sends them to WebRTC app.
+     * @param {MouseEvent} event
+     */
+    _wheel(event) {
+        let data = {
+            "event": "MouseScroll",
+            "x": this.x,
+            "y": this.y,
+            "delta_x": -event.deltaX,
+            "delta_y": -event.deltaY,
+        };
+
+        console.error(JSON.stringify(data))
```

Comment:
If you don't mind making sure those new files you check in don't spam the console more than needed that would be nice :)

Reply:
Oops, forgot to remove some debug in there, sorry.